cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java

```diff
@@ -741,6 +741,10 @@ public Boolean apply(AppMds mds) throws Exception {
   @Override
   @Nullable
   public NamespaceMeta createNamespace(final NamespaceMeta metadata) {
+    Preconditions.checkArgument(metadata != null, "Namespace metadata cannot be null.");
+    Preconditions.checkArgument(metadata.getId() != null, "Namespace id cannot be null.");
+    Preconditions.checkArgument(metadata.getName() != null, "Namespace name cannot be null.");
+    Preconditions.checkArgument(metadata.getDescription() != null, "Namespace description cannot be null.");
```

Comment:
Do you need all these check here? Shouldn't it be checked when creating the `NamespaceMeta` already? (through the Builder?)